app/models/call_session.rb

```diff
@@ -1,33 +1,53 @@
 class CallSession
-  KEY = 'CALL-SESSION'
   EXPIRE_CALL_SESSION_IN = 24.hours
+  COUNTRY = 'IN'
 
-  def initialize(user_phone_number, patient_phone_number)
-    @user = User.find_by_phone_number(user_phone_number)
+  attr_reader :patient_phone_number, :user
+
+  def initialize(call_id, user_phone_number, patient_phone_number)
+    @call_id = call_id
+    @user = User.find_by_phone_number(sanitized_phone_number(user_phone_number))
     @patient_phone_number = PatientPhoneNumber.find_by_number(patient_phone_number)
   end
 
-  def save(call_id)
-    Rails.cache.write(session_key(call_id),
+  def authorized?
+    @user.present? &&
+      patient_phone_number.present? &&
+      @user.phone_number != patient_phone_number.number
+  end
+
+  def save
+    Rails.cache.write(CallSession.session_key(@call_id),
                       session_data,
                       expires_in: EXPIRE_CALL_SESSION_IN)
   end
 
-  def authorized?
-    @user.present? &&
-      @patient_phone_number.present? &&
-      @user.phone_number != @patient_phone_number.number
+  def kill
+    Rails.cache.delete(CallSession.session_key(@call_id))
+  end
+
+  class << self
+    def fetch(call_id)
+      data = Rails.cache.fetch(session_key(call_id))
+
+      CallSession.new(call_id,
+                      data[:user_phone_number],
+                      data[:patient_phone_number]) if data.present?
+    end
+
+    def session_key(call_id)
+      [name, call_id].join('/')
```

Comment:
What is `name`?

Reply:
Hm, maybe this should be `self.name` for clarity. This is a class method you can call for any class that gets you the name of the class in String.